File: 2958. Length of Longest Subarray With at Most K Frequency.py
class Solution:
    def maxSubarrayLength(self, nums: List[int], k: int) -> int:
        freq = {}
        left, answer = 0, 0
        for right in range(len(nums)):
            if nums[right] not in freq:
                freq[nums[right]] = 0
            freq[nums[right]] += 1
            while left <= right and freq[nums[right]] > k:
                freq[nums[left]] -= 1
                left += 1
            answer = right - left + 1 if answer < right - left + 1 else answer
        return answer

File: 2958. Length of Longest Subarray With at Most K Frequency.go
func maxSubarrayLength(nums []int, k int) int {
	freq := make(map[int]int)
	left := 0
	answer := 0
	for right := 0; right < len(nums); right++ {
		curr := nums[right]
		if _, ok := freq[curr]; !ok {
			freq[curr] = 0
		}
		freq[curr] += 1
		for left <= right && freq[curr] > k {
			freq[nums[left]] -= 1
			left += 1
		}
		answer = max(answer, right-left+1)
	}
	return answer
}